contracts/strategies/BenqiStrategyForLP.sol

```diff
@@ -0,0 +1,204 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.7.0;
+
+import "../YakStrategy.sol";
+import "../interfaces/IBenqiStakingContract.sol";
+import "../interfaces/IPair.sol";
+import "../lib/DexLibrary.sol";
+
+/**
+ * @notice Benqi strategy for staking QI-AVAX.
+ */
+contract BenqiStrategyForLP is YakStrategy {
+    using SafeMath for uint;
+
+    IBenqiStakingContract private stakingContract;
+    IERC20 private wavaxRewardToken;
+    IERC20 private qiRewardToken;
+    IPair private swapPairToken; // WAVAX-QI LP
+
+    constructor (
+        string memory _name,
+        address _depositToken,
+        address _wavaxRewardToken,
+        address _qiRewardToken,
+        address _stakingContract,
+        address _swapPairToken,
+        address _timelock,
+        uint _minTokensToReinvest,
+        uint _adminFeeBips,
+        uint _devFeeBips,
+        uint _reinvestRewardBips
+    ) {
+        name = _name;
+        depositToken = IERC20(_depositToken);
+        wavaxRewardToken = IERC20(_wavaxRewardToken);
+        qiRewardToken = IERC20(_qiRewardToken);
+        stakingContract = IBenqiStakingContract(_stakingContract);
+        devAddr = msg.sender;
+
+        assignSwapPairSafely(_swapPairToken);
+        setAllowances();
+        updateMinTokensToReinvest(_minTokensToReinvest);
+        updateAdminFee(_adminFeeBips);
+        updateDevFee(_devFeeBips);
+        updateReinvestReward(_reinvestRewardBips);
+        updateDepositsEnabled(true);
+        transferOwnership(_timelock);
+
+        emit Reinvest(0, 0);
+    }
+
+    /**
+     * @notice Initialization helper for Pair deposit tokens
+     * @dev Checks that selected Pairs are valid for trading deposit tokens
+     * @dev Assigns values to swapPairToken0 and swapPairToken
+     */
+    function assignSwapPairSafely(address _swapPairToken) private {
+        require(_swapPairToken > address(0), "Swap pair is necessary but not supplied");
+        swapPairToken = IPair(_swapPairToken);
+        require(address(wavaxRewardToken) == swapPairToken.token0(), "Swap pair does not match WAVAX reward");
```

Comment:
I think it's a good idea to require that it's equal token0 or token1, that's because token order on the pair might change, and although it's deterministically(based on which one is smaller than the other), we can have this "generically" implemented already, as it is on other LP strategies, like joe and DexV6